src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -200,11 +240,30 @@ public ActionRuleDTO updateActionRule(
               new ParameterizedTypeReference<List<ActionRuleDTO>>() {});
     } catch (HttpClientErrorException e) {
       if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-        return null;
+        throw new CTPException(
+            Fault.SYSTEM_ERROR, String.format("Action Plan %s not found", actionPlanId));
       }
       throw e;
     }
 
     return response.getBody();
   }
+
+  public List<ActionPlanDTO> getActionPlansByCollectionExerciseId(String collectionExerciseId)
+      throws CTPException {
+    final MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+    queryParams.add(SELECTOR_COLLECTION_EXERCISE_ID, collectionExerciseId);
+    final ResponseEntity<List<ActionPlanDTO>> responseEntity =
+        getActionPlansByQueryParameters(queryParams);
+
+    List<ActionPlanDTO> actionPlans = responseEntity.getBody();
+
+    if (actionPlans == null) {
+      throw new CTPException(
```

Comment:
again... question mark over whether we should leave the behaviour of deciding what to do on 404 to the caller for consistency

Reply:
woops this function is not getting used anymore, but the reason was the same as above ^^